### PR TITLE
added questions for job post. ability to create them in the new post.…

### PR DESCRIPTION
**Pull request by DavidSurkov**
I know there are lots of mistakes. but I've been looking at this code for too long. And I need some criticism

### CODE REVIEW

File: src/common/pipes/parse-form-data-json.pipe.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+  transform(value: any, _metadata: ArgumentMetadata) {
+    const { except } = this.options;
+    const serializedValue = value;
+    const originProperties = {};
+    if (except?.length) {
+      _.merge(originProperties, _.pick(serializedValue, ...except));
+    }
+    const deserializedValue = deepParseJson(value);
+    //console.log(`deserializedValue`, deserializedValue);
```
Comment (xx-TOOL-xx) [inline src/common/pipes/parse-form-data-json.pipe.ts]: remove commented code

File: src/modules/job-post/dto/create-job-post.dto.ts [hunk 2]
```diff
@@ -3,65 +3,75 @@ import {
...
+  @IsInt()
   readonly categoryId: Category;
 
-  @IsNumber()
   @ApiProperty()
   @Type(() => Number)
+  @IsNumber()
   readonly rate: number;
 
   //@IsArray({ each: true })
```
Comment (xx-TOOL-xx) [inline src/modules/job-post/dto/create-job-post.dto.ts]: remove commented code

File: src/modules/job-post/entities/freelancer-answer.entity.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import {
+  Column,
+  Entity,
+  JoinTable,
+  ManyToOne,
+  PrimaryGeneratedColumn,
+} from 'typeorm';
+import { JobPostQuestion } from './job-post-question.entity';
+import { Freelancer } from '../../freelancer/entities/freelancer.entity';
```
Comment (xx-TOOL-xx) [inline src/modules/job-post/entities/freelancer-answer.entity.ts]: absolute path. The same for all next files

File: src/modules/job-post/job-post.controller.ts [hunk 4]
```diff
@@ -53,7 +53,7 @@ export class JobPostController {
     return this.jobPostService.create(createJobPostDto, null);
   }
 
-  @UseGuards(JwtAuthGuard)
+  //@UseGuards(JwtAuthGuard)
```
Comment (xx-TOOL-xx) [inline src/modules/job-post/job-post.controller.ts]: uncomment

Comment (DavidSurkov) [reply src/modules/job-post/job-post.controller.ts]: done

File: src/modules/job-post/entities/freelancer-answer.entity.ts (hunk 3 above)
Comment (DavidSurkov) [reply src/modules/job-post/entities/freelancer-answer.entity.ts]: done

File: src/modules/job-post/dto/create-job-post.dto.ts (hunk 2 above)
Comment (DavidSurkov) [reply src/modules/job-post/dto/create-job-post.dto.ts]: done

File: src/common/pipes/parse-form-data-json.pipe.ts (hunk 1 above)
Comment (DavidSurkov) [reply src/common/pipes/parse-form-data-json.pipe.ts]: done

**Pull request merged**